src/pkg/cli/debug.go

```diff
@@ -72,23 +71,69 @@ func Debug(ctx context.Context, c client.Client, etag, folder string, services [
 	return nil
 }
 
-func findMatchingFiles(folder string, patterns []string) []*defangv1.File {
+func readFile(path string) *defangv1.File {
+	b, err := os.ReadFile(path)
+	if err != nil {
+		term.Debug("failed to read file:", err)
+		return nil
+	}
+	return &defangv1.File{
+		Name:    filepath.Base(path),
+		Content: string(b),
+	}
+}
+
+func getServices(project *types.Project, names []string) types.Services {
+	// project.GetServices(…) aborts if any service is not found, so we filter them out ourselves
+	if len(names) == 0 {
+		return project.Services
+	}
+	services := types.Services{}
+	for _, s := range names {
+		if svc, err := project.GetService(s); err != nil {
+			term.Debug("skipped for debugging:", err)
+		} else {
+			services[s] = svc
+		}
+	}
+	return services
+}
+
+func findMatchingProjectFiles(project *types.Project, services []string) []*defangv1.File {
 	var files []*defangv1.File
-	fileCount := 0
 
-	err := filepath.Walk(folder, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			term.Debug("error accessing path:", err)
-			return nil // continue walking
+	for _, path := range project.ComposeFiles {
+		if file := readFile(path); file != nil {
+			files = append(files, file)
 		}
+	}
+
+	for _, service := range getServices(project, services) {
+		if service.Build != nil {
+			files = append(files, findMatchingFiles(service.Build.Context, service.Build.Dockerfile)...)
+		}
+		// TODO: also consider other files, lke .dockerignore, .env, etc.
+	}
+
+	return files
+}
+
+func findMatchingFiles(folder, dockerfile string) []*defangv1.File {
+	var files []*defangv1.File
+	patterns := []string{"*.js", "*.ts", "*.py", "*.go", "requirements.txt", "package.json", "go.mod"}
```

Comment:
Could make this a global constant. We will need to update for any new languages we support.